### PR TITLE
[Tests-Only] add tests for sharing the shares_folder

### PR DESCRIPTION
**Pull request by dpakach**
## Description

Add test for sharing share_folder

## Related Issue

- Related https://github.com/owncloud/web/issues/5152

## How Has This Been Tested?

:robot:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] ...

### CODE REVIEW

File: tests/acceptance/features/webUISharingPublicBasic/publicLinkCreate.feature [hunk 1]
```diff
@@ -189,4 +189,18 @@ Feature: Create public link shares
...
-      """
\ No newline at end of file
+      """
+
+
+  Scenario: Sharing the share_folder as public link is not possible
+    Given the setting "shareapi_auto_accept_share" of app "core" has been set to "no"
+    And the administrator has set the default folder for received shares to "Shares"
+    Given user "Brian" has been created with default attributes and without skeleton files
+    Given user "Brian" has created folder "simple-folder"
```
Comment (SwikritiT) [inline tests/acceptance/features/webUISharingPublicBasic/publicLinkCreate.feature:199]:
```suggestion
    And user "Brian" has been created with default attributes and without skeleton files
    And user "Brian" has created folder "simple-folder"
```

File: tests/acceptance/stepDefinitions/sharingContext.js [hunk 2]
```diff
@@ -880,6 +880,16 @@ Then('{string} {string} should be listed in the autocomplete list on the webUI',
   )
 })
 
+Then(
+  'the share permission denied message should be displayed in the sharing dialog on the webUI',
+  async function() {
+    const permissionDeniedMessage = "You don't have permission to share this folder."
```
Comment (kiranparajuli589) [inline tests/acceptance/stepDefinitions/sharingContext.js:886]: why not getting this from feature?

Comment (dpakach) [reply tests/acceptance/stepDefinitions/sharingContext.js:886]: that part of the ui doens't shows any other type of message. Its only for the permission denied (as far as I have seen) so it makes more sense to make specific step for that.

**Review by SwikritiT — COMMENTED** [review]
LGTM :+1:

**Review by individual-it — APPROVED** [review]

**Review by phil-davis — APPROVED** [review]

**Comment by phil-davis** [comment]
@LukasHirt @kulmann this is ready. Please merge it if you like, or wait until after #5018

**Review by LukasHirt — APPROVED** [review]
IMHO fine to merge this 👍

**Pull request merged**